### ISSUE TITLE
Improve UWP scripting workflow 

### ISSUE THREAD

**Issue by gwendalbroudinpro** [Microsoft/MixedRealityToolkit-Unity#2882]
**Does this affect the legacy HoloToolkit (master) or the Mixed Reality Toolkit (mrtk_release)?**
HoloToolkit and Mixed Reality Toolkit

Hello,

On Unity 2018, .NET scripting back end for UWP is deprecated. We now have to use IL2CPP scripting back end.

Because of that, we don't have a way to have IntelliSense and other tooling for UWP C# code (inside UNITY_WSA compilation symbols), since the dll and the compilation symbols are not available in the unity project. Is there a way to generate some kind of "scripting only project" with the proper .sln/csproj/dll files in order to add UWP targeted code more easily ?

I don't have any experience with msbuild stuff but if you know how we could reproduce Unity's UWP C# project generation, I'll be glad to implement it =).

**Comment by StephenHodgson** [comment]
You should be able to open the unity project's solution and get intellesense on Windows namepace symbols as long as you have il2cpp set.

There's no need to add an additional project.

As for debugging, unity is trying to get their managed debugger working so you can attach to a running app on a HoloLens or device to make that easier, but I think it's still some ways off.

**Comment by StephenHodgson** [comment]
I don't think using .NET would fix anything (and would likely make some things worse).

Yes it's easier to debug the solution with it, but that's the only pro in a long list of cons.

As it stands you _should_ be able to get intellesense and all the required UWP tooling you need in the Unity Generated Project Solution (so long as you're using IL2CPP. You won't get this if you're using .NET)

1. Make sure you've switched to the Windows Universal Build in Unity's Build Window.
![image](https://user-images.githubusercontent.com/13334553/46563778-40b5e400-c8b8-11e8-997d-9e31fcd2250a.png)
2. Make sure you're using the 4.x equivalent scripting runtime version.
![image](https://user-images.githubusercontent.com/13334553/46563800-60e5a300-c8b8-11e8-9c2c-6737b0367a16.png)
3. Make sure your Unity Project solution is generating the `*Player` project in your solution explorer.
![image](https://user-images.githubusercontent.com/13334553/46563847-a6a26b80-c8b8-11e8-8403-cd766af4266f.png)
4. Switch to the player project using the dropdown in the class you're currently editing
![image](https://user-images.githubusercontent.com/13334553/46563864-bfab1c80-c8b8-11e8-8a9d-b566fff80eeb.png)
5. Use the `UNITY_WSA` or `WINDOWS_UWP` symbol defines along with `!UNITY_EDITOR`
6. ??? Profit

Although, you'll still need to build and run the windows stuff outside of the editor, as long as you're not working with HoloLens specific stuff (for example you could just test by installing it locally and running in WMR) you should be okay.

**Comment by StephenHodgson** [comment]
Also, this question isn't really specific to MRTK, but a UWP question in general. We're not Unity, nor affiliated with Unity in any way. It's up to them how they want to continue support for UWP specific tooling and such. I think if you want to make that suggestion, I'd post over on the Unity Forums. 

That being said. I too miss .NET, but after seeing what the IL2CPP features enable, it's pretty easy to see why they chose to go down that route.

**Comment by Alexees** [comment]
@StephenHodgson would you mind linking an explanation of what the *.Player projects are, where you know from that these are necessary? I couldn't find anything about them.

**Comment by gwendalbroudinpro** [comment]
@StephenHodgson I've never saw this *.player csproj file either. I will post a message on Unity's forum then, thanks for your help ! 
@Alexees if I find anything, I will let you know.

**Comment by gwendalbroudinpro** [comment]
Okay @Alexees I found how to generate the Player csproj file. 
1. Install VS 2017 with [unity tools](https://visualstudio.microsoft.com/fr/vs/unity-tools/)
2. Set VS 2017 as your external script editor, [instructions](https://docs.microsoft.com/fr-fr/visualstudio/cross-platform/getting-started-with-visual-studio-tools-for-unity?view=vs-2017)  
3. Close Unity 
4. Delete all .sln and .csproj files and Library and obj folder in your Unity project
5. Reopen unity and click on `open C# project` in assets view, it will generate all sln and csproj files 

I've tested in in unity 2018.1.9f2
